### ISSUE TITLE
KubeletConfiguration doesn't work while using ntpConfiguration

### ISSUE THREAD

**Issue by kobin23** [aws/eks-anywhere#8594]
**What happened**: Kubelet configuration didn't apply to kubernetes nodes, when ntp configuration used in VSphereMachineConfig manifests of these nodes

**What you expected to happen**: Kubelet configuration applied correctly along with configured ntp servers

**How to reproduce it (as minimally and precisely as possible)**: Set any settings in kubeletConfiguration for controlPlane or workerNode (ex. maxPods), set any valid servers for ntp in VSphereMachineConfig manifests of nodes, for which kubeletConfiguration used

**Anything else we need to know?**: Bottlerocket OS used for cluster

**Environment**:
- EKS Anywhere Release: v.0.20.2

**Comment by mitalipaygude** [comment]
Thank you for creating this issue. Do you mind adding more details about the cluster configuration?

Was this a create or an upgrade? If it was an upgrade, what version was the cluster originally created with? What was the kubernetes version used?
Without divulging sensitive information, do you mind sharing the cluster spec used? Specifically the `hostOSConfiguration` and the `kubeletConfiguration` specified?

In the `hostOSConfiguration` was the field for `bottlerocketConfiguration` also populated with any settings?

When the cluster was created, did the NTP server settings get set properly?

**Comment by kobin23** [comment]
Hi, @mitalipaygude !

>Was this a create or an upgrade? If it was an upgrade, what version was the cluster originally created with? What was the kubernetes version used?

I tried both creating new cluster, and upgrading freshly created cluster, behavior was the same in both scenarious. Every time I add `ntpConfiguration` to `VSphereMachineConfig`, `kubeletConfiguration` get ignored. Kubernetes version was 1.30.


> In the hostOSConfiguration was the field for bottlerocketConfiguration also populated with any settings?

No, only `ntpConfiguration`


> When the cluster was created, did the NTP server settings get set properly?`

Yes, NTP settings was correct and got applied to nodes

Example of config with both `kubeletConfiguration` and `hostOSConfiguration`, all other parts which not related to this problem removed:
```
apiVersion: anywhere.eks.amazonaws.com/v1alpha1
kind: Cluster
metadata:
  name: workload-test
spec:
  controlPlaneConfiguration:
    count: 3
    kubeletConfiguration:
      kind: KubeletConfiguration
      maxPods: 50
    machineGroupRef:
      kind: VSphereMachineConfig
      name: workload-test-cp
---
apiVersion: anywhere.eks.amazonaws.com/v1alpha1
kind: VSphereMachineConfig
metadata:
  name: workload-test-cp
spec:
  osFamily: bottlerocket
  hostOSConfiguration:
    ntpConfiguration:
      servers:
        - ntp.example.loc

```

**Comment by mitalipaygude** [comment]
Hello @kobin23 , thank you for the response. We are looking into this issue and mostly should have a fix in the upcoming patch release for the same.

**Comment by kobin23** [comment]
Fixed in v0.20.4 version
